package peptide_hash_database;

message phd_protein {
   required string            phdpro_name = 1;
   required int32             phdpro_id = 2;
   required string            phdpro_pepseq = 3;
}

message phd_peptide {
   required int32             phdpep_protein_id = 1;
   required int32             phdpep_pepstart = 2;
   required int32             phdpep_peplen = 3;
   required int32             phdpep_cleavedpep = 4; // This should be an enum; left, right, semi, full
}

message phd_parameters {
   required string            phdparam_precut_amino = 1;
   required string            phdparam_postcut_amino = 2;
   required string            phdparam_prenocut_amino = 3;
   required string            phdparam_postnocut_amino = 4;
   required int32             phdparam_missed_cleavage = 5;
   required int32             phdparam_semi_tryptic = 6;
}

message phd_header {
   optional int32             phdhdr_version = 1;
   optional string            phdhdr_protein_source_filename = 2;
   optional string            phdhdr_protein_source_file_digest = 3;
   optional int32             phdhdr_num_proteins = 4;
   optional string            phdhdr_hash_file_name = 5;
   optional string            phdhdr_hash_file_digest = 6;
   required phd_parameters    phdhdr_params = 7;
}

message phd_peptide_mass {
   required int32             phdpmass_mass = 1;
   repeated phd_peptide       phdpmass_peptide_list = 2;
}

message phd_file {
   optional phd_header        phdhdr = 1;
   repeated phd_protein       phdpro = 2;
   repeated phd_peptide_mass  phdpepm = 3;
}
